src/objective-c/GRPCClient/GRPCCall.m

```diff
@@ -422,33 +430,55 @@ - (void)startWithWriteable:(id<GRXWriteable>)writeable {
   // that the life of the instance is determined by this retain cycle.
   _retainSelf = self;
 
-  _responseWriteable = [[GRXConcurrentWriteable alloc] initWithWriteable:writeable
-                                                           dispatchQueue:_responseQueue];
-
-  _wrappedCall = [[GRPCWrappedCall alloc] initWithHost:_host serverName:_serverName path:_path];
-  NSAssert(_wrappedCall, @"Error allocating RPC objects. Low memory?");
-
-  [self sendHeaders:_requestHeaders];
-  [self invokeCall];
-
-  // TODO(jcanizales): Extract this logic somewhere common.
-  NSString *host = [NSURL URLWithString:[@"https://" stringByAppendingString:_host]].host;
-  if (!host) {
-    // TODO(jcanizales): Check this on init.
-    [NSException raise:NSInvalidArgumentException format:@"host of %@ is nil", _host];
-  }
   __weak typeof(self) weakSelf = self;
-  _connectivityMonitor = [GRPCConnectivityMonitor monitorWithHost:host];
-  void (^handler)() = ^{
+  void (^performCall)() = ^{
     typeof(self) strongSelf = weakSelf;
     if (strongSelf) {
-      [strongSelf finishWithError:[NSError errorWithDomain:kGRPCErrorDomain
-                                                      code:GRPCErrorCodeUnavailable
-                                                  userInfo:@{ NSLocalizedDescriptionKey : @"Connectivity lost." }]];
+      strongSelf->_responseWriteable = [[GRXConcurrentWriteable alloc] initWithWriteable:writeable
+                                                                           dispatchQueue:strongSelf->_responseQueue];
+
+      strongSelf->_wrappedCall = [[GRPCWrappedCall alloc] initWithHost:strongSelf->_host
+                                                            serverName:strongSelf->_serverName
+                                                                  path:strongSelf->_path];
+      NSAssert(_wrappedCall, @"Error allocating RPC objects. Low memory?");
+
+      [strongSelf sendHeaders:_requestHeaders];
+      [strongSelf invokeCall];
+
+      // TODO(jcanizales): Extract this logic somewhere common.
+      NSString *host = [NSURL URLWithString:[@"https://" stringByAppendingString:strongSelf->_host]].host;
+      if (!host) {
+        // TODO(jcanizales): Check this on init.
+        [NSException raise:NSInvalidArgumentException format:@"host of %@ is nil", strongSelf->_host];
+      }
+      strongSelf->_connectivityMonitor = [GRPCConnectivityMonitor monitorWithHost:host];
+      void (^handler)() = ^{
+        typeof(self) strongSelf = weakSelf;
```

Comment:
You're hiding `strongSelf` here with another `strongSelf`. It might be clearer to put all of this in a method, and on line 437 do:
```obj-c
if (strongSelf) {
  [strongSelf someMethod];
}
```
Then you can use `self` inside that method, and refer to ivars normally (`_myIvar` instead of `strongSelf->_myIvar`).